systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -1957,6 +1956,14 @@ void setup() throws Exception {
     @Override
     protected void tearDownEnvironmentAfterEach() throws Exception {
         super.tearDownEnvironmentAfterEach();
+        if (cluster.getListOfDeployedResources().contains(TEMPLATE_PATH)) {
+            cluster.deleteCustomResources(TEMPLATE_PATH);
+        }
+
+        kubeClient().listPods().stream()
+            .filter(p -> p.getMetadata().getName().startsWith(OPENSHIFT_CLUSTER_NAME))
+            .forEach(p -> PodUtils.deletePodWithWait(p.getMetadata().getName()));
```

Comment:
Why is this moved from the one scenario where everything is deployed (and deleted) to afterEach?

Reply:
If the test fails for example in Kafka creation (deploying), the resources are not deleted properly. This will ensure that all resources from the `testDeployKafkaClusterViaTemplate` will be removed and will not affect other tests.

Reply:
I was looking for some solution to be executed only for the `testDeployKafkaClusterViaTemplate()`, but there is no annotation for that. Only thing what I can add is the `try{}catch{}` block and that's not good approach from MPOV

Reply:
Oh make sense to me!

Reply:
Maybe we can talk to someone from JUnit to add anno to do something after specific test :smile:

Reply:
@After if testName = mySpecific test 
https://stackoverflow.com/a/41321322 ;-)